[Fluffykan] iP

Fluffykan opened the pull request:
# Slave
> “I choose a lazy person to do a hard job. Because a lazy person will find an easy way to do it.” - Bill Gates

## With Slave, remembering things has never been easier, because you don't have to remember them anymore!
## Additionally, its:
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

## All you need to do is,
1. Download it [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html)
1. double click it
1. add your tasks
1. let it manage your tasks for you 😄 

And its **FREE**!

Features:

- [x] Managing Tasks
- [x] Managing Deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:


```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Fluffykan committed:
level-0 complete

Renamed bot,
Added greet and exit functionality

Fluffykan committed:
level-0

chatbot will greet and exit

Fluffykan committed:
level-0

bot only greets and exits

Fluffykan committed:
level-1

slave will echo user's input

Fluffykan committed:
level-2

slave can now add items to list, and list out the items in the list

Fluffykan committed:
level-3

slave can now mark tasks as done and not done

Fluffykan committed:
level-3

bug fix for issue with mark and unmark function not parsing string correctly

Fluffykan committed:
level-4

created Todo, Deadline, Event classes

Fluffykan committed:
ATextUiTesting

modified runtest.bat such that it can be run successfully

Fluffykan committed:
level-6

delete feature implemented

Fluffykan committed:
level-0 refactor

refactored code as per coding standards:

* moved all files into a package
* listed all import files instead of importing all
* removed indentation for switch case

Fluffykan committed:
levels 1 through 6 refactoring

refactored code as per coding standards:

* moved all files into a package
* listed all import files instead of importing all
* removed indentation for switch case

Fluffykan committed:
level-7

save and load method implemented
save() method is called at the end of every of the following method calls:

*addToList()
*deleteTask()
*markAsDone()
*markAsIncomplete()

load method is called at the start of the program
in the event of a missing save file, program will skip over the load() method

Fluffykan committed:
level-7

bug fix for where loading of tasks from save file in previous commit
did not pull the full string after /from and /to for events, due to the assumption
that there is only one string without spacing after each identifier

Fluffykan committed:
level-7

commiting in a new branch instead of on main

Fluffykan committed:
added savefile.txt to .gitignore

Fluffykan committed:
Merge branch 'master' into level-7

Fluffykan committed:
post reset from wrong merge

Fluffykan committed:
savefile.txt

commit does not affect execution of program

Fluffykan committed:
undo level-7 changes

commented out level-7 changes
trying to create new branch, commit on that branch, then merge into main

Fluffykan committed:
level-7

added changes required for level 7

Fluffykan committed:
undo level-7 changes again

Fluffykan committed:
level-7

pushing level-7 changes

Fluffykan committed:
undo level-7 changes again

Fluffykan committed:
delete savefile.txt

Fluffykan committed:
bug fix for toString in Event class

it was previously missing a space which resulted in errors when calling the load() method

Fluffykan committed:
Merge branch 'branch-Level-7'

Fluffykan committed:
level-8

implemented the changes necessary for reading dates after /from, /by, and /to

Fluffykan committed:
level-8 features complete

Deadline and Events now store a LocalDate instead of a String

keyword "schedule (date in yyyy-mm-dd format)" displays all deadlines due after the date,
and events happening on the date

"schedule (date)" is able to handle bad inputs, eg. "schedule ", "schedule", "schedule (date)",
where the date does not follow the format yyyy-mm-dd.
All bad inputs will be rejected and no Tasks will be printed

Fluffykan committed:
undo ff merge from branch-level-8

Fluffykan committed:
Merge branch 'master' of https://github.com/Fluffykan/ip

Fluffykan committed:
complete doc comments for scheduleOn

Fluffykan committed:
Merge branch 'branch-Level-8'

Fluffykan committed:
level-8 refactor with OOP

Slave methods broken down into 3 classes:

*Ui
*Storage
*Parser

functionality unchanged

Fluffykan committed:
Merge tag 'A-MoreOOP'

Fluffykan committed:
bug fix

fix issue where new file and directory is not created if the path does not exist

fix issue where "bye" does not terminate the program

Fluffykan committed:
Merge branch 'A-MoreOOP'

Fluffykan committed:
Merge branch 'add-gradle-support'

Fluffykan committed:
Setup Gradle to be able to build and run app

Fluffykan committed:
Create unit testing using junit

Create testcases for Storage and Parser classes

Refactor Slave, Parser, and Ui class methods to enable unit testing on
the individual class methods by changing return type of some methods
from void to boolean

Fluffykan committed:
Merge branch 'A-JUnit'

Fluffykan committed:
Package App as a JAR file

Fluffykan committed:
Merge tag 'A-Jar'

Fluffykan committed:
Modify some of the existing header comments

Fluffykan committed:
Modify code to fit coding standard

Fluffykan committed:
Merge branch 'branch-A-JavaDoc'

Fluffykan committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/slave/Task.java

Fluffykan committed:
Create find() function as per requirement

Fluffykan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/slave/Parser.java

The0nlyJuan commented on src/main/java/slave/InvalidChronologicalOrderException.java:
```diff
@@ -0,0 +1,7 @@
+package slave;
```
Its good that it handles date that have invalid chronological order too!

The0nlyJuan reviewed (commented):
LGTM!

Fluffykan committed:
add checkstyle plugin to gradle

fix all issues raised by checkstyle

Fluffykan committed:
Merge branch 'A-Checkstyle'

siriousguy commented on src/main/java/slave/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+        this.deadline = deadline;
+    }
+
+    public String getDeadline() {
+        return this.deadline.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+    }
+
+    public LocalDate getRawDeadline() {
+        return this.deadline;
+    }
```
A method comment will be useful to distinguish between these two

So getDeadline can be getFormattedDeadline

siriousguy commented on src/main/java/slave/Pair.java:
```diff
@@ -0,0 +1,37 @@
+    private U u;
+
+    /**
+     * creates a pair containing values t of type T and u of type U
```
Nitpicking here but "Creates" just keeps the Javadocs more consistent with the rest of your naming.

siriousguy commented on src/main/java/slave/Parser.java:
```diff
@@ -0,0 +1,345 @@
+    }
+
+    /**
+     * prompts the user for input
```
Might want to capitalise "prompts".

siriousguy commented on src/main/java/slave/Parser.java:
```diff
@@ -0,0 +1,345 @@
+    }
+
+    /**
+     * marks the task as completed by changing the boolean value to true
```
Might want to capitalise "marks" too 😅

siriousguy commented on src/main/java/slave/Parser.java:
```diff
@@ -0,0 +1,345 @@
+    }
+
+    /**
+     * removes the Task from list
```
capitalise "removes"

siriousguy commented on src/main/java/slave/Parser.java:
```diff
@@ -0,0 +1,345 @@
+        2 - Event
+         */
+        boolean didInsert = true;
+        try {
+            switch (i) {
+            case 0:
+                // todo
+                //@@ author Carl Smotricz -reused
+                // String.trim() method used to remove all whitespaces from a string to check if the string only
+                // contains whitespaces is reused from the StackOverflow reply
+                if (s.isEmpty() || s.trim().isEmpty()) {
+                    //@@author
+                    throw new InvalidTaskFormatException("No task descriptor");
+                }
+                list.add(new Todo(s));
+                break;
+            case 1:
+                // deadline
```
Good use of comments here!

siriousguy commented on src/main/java/slave/Parser.java:
```diff
@@ -0,0 +1,345 @@
+            System.out.println("Should have known better than to waste time trying to help you...");
+        }
+    }
+
+
```
Notice two line separation here, might want to remove one of the lines.

siriousguy commented on src/main/java/slave/Storage.java:
```diff
@@ -0,0 +1,164 @@
+    }
+
+    /**
+     * converts the taskList to a string format and writes it to the savefile
+     * every line contains only 1 task
+     * string format is as per the return value of toString() method of the respective task
+     * creates a new file at the specified address called "savefile.txt" in the event of a missing save file
```
Can make use of some capitalisation here.

I.. shall not mention capitalisation anymore

siriousguy reviewed (commented):
LGTM! I really enjoyed reading the interactions your bot will have. Just have to work on some code commenting consistency such as capitalisation.

side note: the java coding standard did mention punctuation at the end of the comment such as ".'; since there is a consistent lack of punctuation throughout the comments, I think it might be ok ?

- Punctuation behind each parameter description.

ethanwangkangen commented on src/main/java/slave/Parser.java:
```diff
@@ -0,0 +1,345 @@
+    /**
+     * Prints the user's input
+     *
+     * @param s is the user's input
```
I think the coding standard wants punctuation at the end of each parameter?

ethanwangkangen commented on src/test/java/slave/ParserTest.java:
```diff
@@ -0,0 +1,97 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ParserTest {
```
Good use of extensive test cases! Very comprehensive

ethanwangkangen commented on src/main/java/slave/Task.java:
```diff
@@ -0,0 +1,78 @@
+    private String task;
+
+    /**
+     * creates a task of name task
```
Should capitalise "Creates", and keep the formatting throughout

ethanwangkangen commented on src/main/java/slave/Todo.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    @Override
+    public String save() {
```
Perhaps this could be named something else for clarity?

ethanwangkangen reviewed (commented):
Overall LGTM!

Fluffykan committed:
Complete level 10 tasks

* Implement GUI for Slave.
* Refactor methods to return either String or String[] for Slave to
  be able to print the messages in the GUI.
* Edit testcases to accomodate the changes made during
  refactoring.

Fluffykan committed:
Merge tag 'Level-10'

# Conflicts:
#	src/main/java/slave/Parser.java
#	src/main/java/slave/Slave.java
#	src/main/java/slave/Storage.java
#	src/main/java/slave/Ui.java

Fluffykan committed:
add asserts to code

asserts added to help ensure the correctness of the execution of the code
at runtime to check that the code is working as intended

asserts used in the following files
* Parser.java
* MainWindow.java

Fluffykan committed:
Merge pull request #1 from Fluffykan/A-Assertions

add asserts to code

Fluffykan committed:
Merge remote-tracking branch 'origin/master'

Fluffykan committed:
refactor code by using abstraction

refactoring done for:
* Parser.java
* Storage.java

shortened lengthy code in the files via abstraction

Fluffykan committed:
Merge pull request #2 from Fluffykan/branch-A-CodeQuality

branch-A-CodeQuality

Fluffykan committed:
Merge remote-tracking branch 'origin/master'

Fluffykan committed:
Merge branch 'master' into branch-A-Streams

Fluffykan committed:
Add implementation to support B-recurringTasks

A new abstract class "RecurringTypeTask" was created to add the recurringType parameter to all tasks
that can have recurrences. Deadline and Event classes now extend from RecurringTypeTask

An enum is used to define the recurring type of tasks, with the following possible options:
* Daily
* Weekly
* Monthly
* Bimonthly
* Quarterly
* Biannually
* Annually

The format which tasks are saved in in the save file is also changed to accomodate the addition of the
recurrence type.

Upon launching the application Slave will:
* Highlight the tasks that have ended prior to the day of launching the application.
This list of tasks will not be stored after closing the application.
* Create new recurring tasks for tasks that have ended prior to the day of launching the application
and add them to the task list.
* Remove all tasks that have ended prior to the day of launching the application.

Slave will allow the creation of tasks that have already ended prior to the day of creation, to allow
users to not have to calculate date of its next occurance for recurring tasks.
For non recurring tasks, it does not affect the functionality of the application as it will be removed
from the task list the next time the application is run

Fluffykan committed:
Merge pull request #3 from Fluffykan/BCD-Extension

Bcd extension

Fluffykan committed:
Merge remote-tracking branch 'origin/master'

Fluffykan committed:
Creating Slave object throws Concurrent Modification Exception

Adding and removing Tasks from the task list using a forEach()
method causes Concurrent Modification Exception to be thrown when
trying to identify deadlines or events that have ended

Let's separate the removing of tasks and addition of tasks from
the checking of tasks that have ended to prevent the exception from
being thrown

Fluffykan committed:
No differentiation between good and bad user input

Currently good and bad user inputs have the same appearance.
It is not immediately clear to the user when they have provided a bad
input.

Let's change the appearance of the dialog box when the user provides a
bad input, by changing its color to red.

Fluffykan committed:
Change user profile photo

The user profile picture is the default photo.

Let's change it to a new photo

Fluffykan committed:
Modify parser testcases

Previous refactoring of the markAsDone() and markAsIncomplete() methods
have created bugs in the  testcases.

The affected testcases have been modified to accept the new return type
of a String array.

Fluffykan committed:
Clean up code style using checkstyle

Checkstyle was used to check the code for coding standard violations.

The blocks of code causing the violations have been changed, and the
code now passes the checkstyle audit.

Fluffykan committed:
Add chatbot name to window title

The title bar of the window now displays "Slave" to meet submission
requirements.

Fluffykan committed:
Fix issue with Todo display format

There was an issue with the display format of a Todo Task
where there was no space between the task name and the square
brackets indicating if the task was complete.

A space is added to make it easier for the user to read the task
name.

Fluffykan committed:
Update README.md

Fluffykan committed:
Update README.md

Previous readme update updated the readme in the root directory.

Current readme update updated the readme in /docs directory

Fluffykan committed:
Merge branch 'master' of https://github.com/Fluffykan/ip
